src/main/java/com/shopify/ShopifySdk.java

```diff
@@ -735,11 +736,18 @@ public ShopifyFulfillment createFulfillment(
 	 * @param fulfillmentOrders
 	 *            the fulfillment orders list to create the new fulfillment
 	 * @return the newly created fulfillment
+	 * @throws ShopifyUnsupportedActionException
+	 *             in case we have a fulfillment associated with a
+	 *             fulfillmentOrder without supported action
 	 */
 	public ShopifyFulfillment createFulfillment(
 			final ShopifyFulfillmentCreationRequest shopifyFulfillmentCreationRequest,
-			List<ShopifyFulfillmentOrder> fulfillmentOrders) {
-		return this.createFulfillmentWithFulfillmentOrderApi(shopifyFulfillmentCreationRequest, fulfillmentOrders);
+			List<ShopifyFulfillmentOrder> fulfillmentOrders) throws ShopifyUnsupportedActionException {
+		try {
+			return this.createFulfillmentWithFulfillmentOrderApi(shopifyFulfillmentCreationRequest, fulfillmentOrders);
+		} catch (Exception e) {
+			throw e;
```

Comment:
This try catch isn't adding any context here. You can remove it, and just let the client handle the exception.

Reply:
yea, this was me testing I was having problems catching the exception and forgot this here
tks